Sage/carbon
# refactor(form-field): convert to typescript

### Proposed behaviour

#### Refactor FormField to Typescript

- Converted `form-field.component.js` file to typescript
- Removed `form-field.d.ts` file
- Converted spec files to typescript
- Converted style files to typescript
- Convert index file to typescript and removed `index.d.ts`
- Export prop types from index file

#### Bug fixes
- In `Label`, fix typing of `helpTabIndex` prop to also accept numbers as well as strings  
- Modify typing of `TabContextProps` so props are optional - to reflect that these could be undefined at runtime.
- Remove unused `helpTabIndex` prop from `Switch`, `FormField` and `Label`
- Add missing `id` prop to type definition of `ButtonToggleGroup` (addresses #5314)

### Current behaviour

- `FormField` is written in javascript



### Checklist




#### QA


### Testing instructions

- Double check that no regressions have been accidentally introduced from this refactor.
- Verify if `tabIndex` console warning for components that use `FormField` have been resolved. 
- Verify in the console warning related to the missing id for ButtonToggleGroup has been resolved.

See [this codesandbox example](https://codesandbox.io/s/weathered-wood-tbgnmo?file=/src/index.tsx) which renders all the consumer-facing components that directly use `FormField`.
